compiler/rustc_hir_typeck/src/mem_categorization.rs

```diff
@@ -1,774 +0,0 @@
-//! # Categorization
```

Comment:
Goodbye to this comment. I could possibly pull all of the `fn cat_*` that I moved into `expr_use_visitor.rs` into a separate impl and copy this comment over. Let me know if necessary.